src/plugins/intel_cpu/tests/unit/graph/merge_transpose_reorder_test.cpp

```diff
@@ -0,0 +1,229 @@
+    }
+    DummyNode(const ov::Shape& shape,
+            const ov::element::Type_t& prc,
+            const std::string& name,
+            const std::string& type,
+            const GraphContext::CPtr context) :
+        Node(type, name, context) {
+        // dummy node of the same shape and precision to both input and output.
+        outputShapes.emplace_back(shape);
+        inputShapes.emplace_back(shape);
+        addOriginalOutputPrecision(InferenceEngine::details::convertPrecision(prc));
+        addOriginalInputPrecision(InferenceEngine::details::convertPrecision(prc));
+    }
+
+    void getSupportedDescriptors() override {
+        if (getParentEdges().size() != 1)
+            OPENVINO_THROW("Incorrect number of input edges for layer " + getName());
+        if (getChildEdges().empty())
+            OPENVINO_THROW("Incorrect number of output edges for layer " + getName());
+    }
+
+    void initSupportedPrimitiveDescriptors() override {
+        if (!supportedPrimitiveDescriptors.empty())
+            return;
+
+        NodeConfig config;
+        config.inConfs.resize(1);
+        config.outConfs.resize(1);
+
+        config.inConfs[0].inPlace(m_inplace & Edge::LOOK::LOOK_DOWN ? 0 : -1);
+        config.inConfs[0].constant(false);
+        config.outConfs[0].inPlace(m_inplace & Edge::LOOK::LOOK_UP ? 0 : -1);
+        config.outConfs[0].constant(false);
+
+        auto layoutCreator = BlockedDescCreator::getCommonCreators().at(m_layout);
+        auto& originInputPrecisions = getOriginalInputPrecisions();
+        config.inConfs[0].setMemDesc(layoutCreator->createSharedDesc(originInputPrecisions[0], getInputShapeAtPort(0)));
+        config.outConfs[0].setMemDesc(layoutCreator->createSharedDesc(originInputPrecisions[0], getOutputShapeAtPort(0)));
+
+        supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::undef);
+    };
+
+    void setLayout(LayoutType layout) {m_layout = layout;}
+    void setInplaceDirection(Edge::LOOK look) {m_inplace = look;}
+
+    bool isExecutable() const override {return false;}
+    void execute(dnnl::stream strm) override {};
+    bool created() const override {return true;}
+
+private:
+    LayoutType m_layout = LayoutType::ncsp;
+    Edge::LOOK m_inplace = Edge::LOOK::LOOK_UP;
+};
+} // namespace MergeTransposeReorderCPUTest
+
+using namespace MergeTransposeReorderCPUTest;
+
+/*
+ * MergeTransposeReorderIsOptimizedCPUTest to test the CPU plugin-in MergeTransposeReorder graph optimizer
+ * under the circumstance that the upstream node or downstream node is inPlaced thereby the inserted Reorder
+ * cannot be optimized.
+ */
+class MergeTransposeReorderIsOptimizedCPUTest : public ::testing::Test {
+public:
+    void Validate() const {
+        CheckTransposeCount(0);
+        CheckReorderOptimized(std::string("_fake"), false);  // the fused node is of name "reshape_abcd_acdb_fake"
+    }
+
+    void SetUp() override {
+        CreateGraph();
+    }
+
+protected:
+    /*  graph typology
```

Comment:
Could you please rename `Reshape` -> Dummy and `Multiply` to `Dummy`? The fact that the `Reshape` and `Multiply` types are assigned to the dummy node is only needed to support the node constructor interface.

Reply:
done